js/src/entity.ts

```diff
@@ -116,6 +116,10 @@ export class Entity {
   getCaption(): string {
     for (const property of this.schema.caption) {
       for (const value of this.getProperty(property)) {
+        if (typeof value !== "string" && typeof value.getCaption === "function") {
```

Comment:
This change allows for captions to come from other entities. With trip startLocation, endLocation etc are Address entities so if only these are included without this change an error occurs.